src/java.base/share/classes/java/net/SocketPermission.java

```diff
@@ -1079,8 +1086,8 @@ public boolean equals(Object obj) {
         if (this.invalid || that.invalid)
             return false;
 
-        if (this.cname != null) {
-            return this.cname.equalsIgnoreCase(that.cname);
+        if (this.cnames != null) {
+            return this.cnames[0].equalsIgnoreCase(that.cnames[0]);
```

Comment:
@jaysk1 Can this comparison fail for SocketPermission objects that have multiple cname entries in different orders?